### PR TITLE
Grafana DS: Fix dropzone showing upload file even if there is a file

### PR DESCRIPTION
**Pull request by oscarkilhed**
**What is this feature?**

Fixes an issue where the FileDropzone would show upload instead of replace even if there was a file if the user left edit mode and then returned.

**Who is this feature for?**

Users testing the drag and drop spreadsheet feature

### CODE REVIEW

**Review by baldm0mma — APPROVED** [review]
Great work; any tests need to be updated after this?

File: packages/grafana-ui/src/components/FileDropzone/FileDropzone.tsx [hunk 1]
```diff
@@ -46,6 +46,7 @@ export interface FileDropzoneProps {
    */
   fileListRenderer?: (file: DropzoneFile, removeFile: (file: DropzoneFile) => void) => ReactNode;
   onFileRemove?: (file: DropzoneFile) => void;
+  primaryTextSupplier?: (files?: DropzoneFile[], options?: BackwardsCompatibleDropzoneOptions) => string;
```
Comment (leventebalogh) [inline packages/grafana-ui/src/components/FileDropzone/FileDropzone.tsx:49]: Is there a reason to make the `files` parameter optional? As far as I can see the component sets it to an empty array by default.

File: packages/grafana-ui/src/components/FileDropzone/FileDropzone.tsx [hunk 2]
```diff
@@ -198,7 +207,7 @@ export function FileDropzone({ options, children, readAs, onLoad, fileListRender
     <div className={styles.container}>
       <div data-testid="dropzone" {...getRootProps({ className: styles.dropzone })}>
         <input {...getInputProps()} />
-        {children ?? <FileDropzoneDefaultChildren primaryText={getPrimaryText(files, options)} />}
+        {children ?? <FileDropzoneDefaultChildren primaryText={primaryTextSupplier(files, options)} />}
```
Comment (leventebalogh) [inline packages/grafana-ui/src/components/FileDropzone/FileDropzone.tsx:210]: I think we should add a test case for covering a custom text supplier in `FileDropzone.test.tsx`.

**Review by leventebalogh — APPROVED** [review]
LGTM 👍

File: packages/grafana-ui/src/components/FileDropzone/FileDropzone.tsx (hunk 2 above)
Comment (oscarkilhed) [reply packages/grafana-ui/src/components/FileDropzone/FileDropzone.tsx:210]: You're right. I added a simple test to make sure it actually renders with the text from a passed supplier. Let me know what you think @leventebalogh.

**Pull request merged**